rest/src/main/kotlin/builder/components/CompositeButtonBuilder.kt

```diff
@@ -0,0 +1,8 @@
+package dev.kord.rest.builder.components
+
+import dev.kord.common.annotation.KordPreview
+
+@KordPreview
+interface CompositeButtonBuilder {
+    fun button(builder: ButtonBuilder.() -> Unit)
+}
```

Comment:
Making this non-inline prevents people from calling suspending functions in the lambda, we generally want to avoid that.

Additionally, there's no way to access, modify or delete any of the already created buttons.

Reply:
The problem here is that I can't inline this as it is an interface, which makes this whole builder thing not really ideal either way

Earlier you could add components at the wrong places, but now you can create confusing code.

```kotlin
createMessage {
  components {
    actionRow {
      actionRow {
      }
   }
}
```

Reply:
You can inline them if you expose a mutable list with all stored components, but the issue there is inheritance. What we have is a builder that can accept components of any type (message) and a builder that can accept components of any type except action rows (action rows). 

"Everything but X" is hard to represent in Kotlin's type system, especially if you then have to reintroduce that type, if you'd go that route it's best to split up the types in separate lists. Have action row builders contain a list of button builders, and the message component builder inherit that but also have a list of action row builders in addition. 

I'm generally not a fan of inheritance in builders like this, it's not unreasonable to assume that Discord would introduce a type that suddenly breaks the rules (like something that can only sit in a row but not in at the root of a message's components). That said, if you can make it work I'll approve it, we break the API whenever Discord adds something anyhow.